### PR TITLE
[Docs] add cloudformation backed guide for s3-sqs setup for filebeats

### LINKED ISSUE #40624: Create cloudformation template to setup s3 + sqs
In Filebeat/agent, we can use the aws-s3 input to retrieve logs from S3 objects that are pointed to by S3 notification events read from an SQS queue. In order to avoid manual setup every time by following https://www.elastic.co/guide/en/beats/filebeat/current/filebeat-input-aws-s3.html#_s3_and_sqs_setup, we should create a cloudformation template to:

- create s3
- create sqs
- setup s3+sqs

### PR DESCRIPTION
**Pull request by Kavindu-Dodan**
## Proposed commit message

Improves filebeat S3 -> SQS documentation by adding a detailed guide backed by a CloudFormation template.

## Checklist

- [x] My code follows the style guidelines of this project
~~- [ ] I have commented my code, particularly in hard-to-understand areas~~
- [x] I have made corresponding changes to the documentation
~~- [ ] I have made corresponding change to the default configuration files~~
~~- [ ] I have added tests that prove my fix is effective or that my feature works~~
~~- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.~~

## Related issues

Fixes #40624

## Screenshots

image

### CODE REVIEW

File: x-pack/filebeat/docs/inputs/input-aws-s3.asciidoc [hunk 1]
```diff
@@ -573,10 +573,126 @@ sqs:GetQueueAttributes
...
+----
+filebeat.inputs:
+- type: aws-s3
+  queue_url: <URL_FROM_STACK>
+  expand_event_list_from_field: Records
+  credential_profile_name: elastic-beats
+----
++
+
+With this configuration, filebeat avoids polling and utilize SQS notifications to extract logs from the S3 bucket.
```
Comment (kaiyan-sheng) [inline x-pack/filebeat/docs/inputs/input-aws-s3.asciidoc:695]:
```suggestion
With this configuration, filebeat avoids polling and utilizes SQS notifications to extract logs from the S3 bucket.
```

**Review by kaiyan-sheng — COMMENTED** [review]
The template looks good to me. Should we also add a command for `delete-stack`?

**Comment by Kavindu-Dodan** [comment]
> The template looks good to me. Should we also add a command for `delete-stack`?

Good point. I thought about this but didn't add thatt. I think the end user could continue to use the setup for their validations (ex:- programmatically upload logs or add through console). A stack deletion step could confuse the user 🤔  

Anyway, let me know your thoughts

**Comment by kaiyan-sheng** [comment]
> Good point. I thought about this but didn't add that. I think the end user could continue to use the setup for their validations (ex:- programmatically upload logs or add through console to S3). A stack deletion step could confuse the user 🤔

I dont have a strong opinion on this either. We can leave it out for now and change it later if we get any feedback then!

**Comment by Kavindu-Dodan** [comment]
> > Good point. I thought about this but didn't add that. I think the end user could continue to use the setup for their validations (ex:- programmatically upload logs or add through console to S3). A stack deletion step could confuse the user 🤔
> 
> I dont have a strong opinion on this either. We can leave it out for now and change it later if we get any feedback then!

yeah, I agree. I will merge the PR with your approval :)

**Pull request merged**